File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        sort(nums.begin(), nums.end());

        for (size_t i=0; i<nums.size(); i++) {
            if (nums[i] == nums[i+1]) {
                return nums[i];
            }
        }
        
        return -1;
    }
};

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.py
class Solution(object):
    def findDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        #nums.sort()
        start, end = 0, len(nums)-1
        while start < end:
            if nums[start] == nums[end]:
                break
            elif nums[start] in nums[start+1:]:
                break
            elif nums[end] in nums[:end]:
                start = end
                break
            else:
                start+=1

        return nums[start]